src/Core/Migration/V6_7/Migration1753191228AddMediaThumbnailSizeIdToMediaThumbnail.php

```diff
@@ -0,0 +1,105 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_7;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Migration\MigrationStep;
+use Shopware\Core\Framework\Uuid\Uuid;
+
+/**
+ * @internal
+ */
+#[Package('discovery')]
+class Migration1753191228AddMediaThumbnailSizeIdToMediaThumbnail extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1753191228;
+    }
+
+    public function update(Connection $connection): void
+    {
+        if (!$this->columnExists($connection, 'media_thumbnail', 'media_thumbnail_size_id')) {
+            $this->addColumn($connection, 'media_thumbnail', 'media_thumbnail_size_id', 'BINARY(16)');
+        }
+
+        $this->migrateMediaThumbnailRows($connection);
+
+        $connection->executeStatement('
+            ALTER TABLE `media_thumbnail`
+            MODIFY COLUMN `media_thumbnail_size_id` BINARY(16) NOT NULL
+        ');
+
+        if ($this->indexExists($connection, 'media_thumbnail', 'fk.media_thumbnail.media_thumbnail_size_id')) {
+            return;
+        }
+
+        $connection->executeStatement('
+            ALTER TABLE `media_thumbnail`
+            ADD CONSTRAINT `fk.media_thumbnail.media_thumbnail_size_id`
+            FOREIGN KEY (`media_thumbnail_size_id`)
+            REFERENCES `media_thumbnail_size` (`id`)
+            ON DELETE RESTRICT ON UPDATE CASCADE
+        ');
+
+        $this->registerIndexer($connection, 'media.indexer');
+    }
+
+    private function migrateMediaThumbnailRows(Connection $connection): void
+    {
+        $batchSize = 10000;
```

Comment:
is that a reasonable chunk size? I cannot judge, but usually we only take `1000`